core/src/main/scala/akka/persistence/r2dbc/internal/Sql.scala

```diff
@@ -21,7 +23,19 @@ object Sql {
    * to include a literal `?`. Trims whitespace, including line breaks. Standard string interpolation arguments `$` can
    * be used.
    */
-  implicit class Interpolation(val sc: StringContext)(implicit adapter: QueryAdapter) extends AnyRef {
+  implicit class Interpolation(val sc: StringContext) extends AnyVal {
+    def sql(args: Any*): String =
+      fillInParameterNumbers(trimLineBreaks(sc.s(args: _*)))
+  }
+
+  /**
+   * INTERNAL API: Scala string interpolation with `sql` prefix. Replaces `?` with numbered `\$1`, `\$2` for bind
+   * parameters. Use `??` to include a literal `?`. Trims whitespace, including line breaks. Standard string
+   * interpolation arguments `$` can be used.
+   */
+  @InternalApi private[akka] implicit class InterpolationWithAdapter(val sc: StringContext)(implicit
+      adapter: QueryAdapter)
```

Comment:
`QueryAdapter` is `private[akka]` and therefore I think the `InterpolationWithAdapter` should also be `private[akka]`, otherwise we should also make QueryAdapter public (InternalStableApi)

Note that this is `AnyRef` because of two parameters, instead of `AnyVal`. I'm not sure this idea with the adapter is worth the indirect complexity it adds. Doesn't we override all sql for sqlserver anyway in the SqlServerDao classes? Those could use `@` as the placeholder instead of `?` and then we could change the ordinary `fillInParameterNumbers` to look for `?` and `@` and replace accordingly. I'll might try that in a separate pr.

Reply:
ok, we don't override all sql, which is good. I'll think a little more about this, but this PR can be merged seperately

Reply:
Correct, we do not override all sql statements, which is why this QueryAdaper was introduced.

Reply:
I leave it as this for now. Projections will still be able to use QueryAdapter since it's `private[akka]`.